bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/annotation/ActionOutput.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.automation.annotation;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Output parameter for an action module
+ *
+ * @author Stefan Triller - initial contribution
+ */
+@Repeatable(ActionOutputs.class)
+@Retention(RUNTIME)
+@Target(ElementType.METHOD)
+public @interface ActionOutput {
+
+    String name();
+
+    String type();
```

Comment:
Wouldn't a class make more sense? Strings can contain typos.
And if you ever want to deserialze stuff the class itself would be helpful in OSGi environment where you don't know if the classloader has access to the respective class.
Or am I wong?

Reply:
I discussed that with @kaikreuzer and if we would only allow java classes here, your recommendation makes a lot of sense. However, we would also like to use string like "ItemName", "ThingName" or whatever and a (to be defined) UI could then render a nice input field.